packages/web/components/cards/stake-dashboard.tsx

```diff
@@ -14,63 +15,83 @@ export const StakeDashboard: React.FC<{
+    const t = useTranslation();
+    const { priceStore, chainStore, queriesStore, accountStore } = useStore();
 
-  const price = balance ? priceStore.calculatePrice(balance) : 0;
+    const osmosisChainId = chainStore.osmosis.chainId;
+    const cosmosQueries = queriesStore.get(osmosisChainId).cosmos;
+    const { chainId } = chainStore.osmosis;
+    const account = accountStore.getWallet(chainId);
+    const address = account?.address ?? "";
+    const osmo = chainStore.osmosis.stakeCurrency;
 
-  const icon = (
-    <div className="flex items-center justify-center text-bullish-500">
-      <div className="mr-2 flex self-center">
-        <Icon id="open-book" height="14px" width="14px" />
-      </div>
-      <span className="caption text-sm">{t("stake.learn")}</span>
-    </div>
-  );
+    const { rewards } =
+      cosmosQueries.queryRewards.getQueryBech32Address(address);
 
-  return (
-    <GenericMainCard
-      title={t("stake.dashboard")}
-      titleIcon={icon}
-      titleIconAction="www.google.com"
-      width="45"
-    >
-      <div className="flex w-full flex-row justify-between py-10">
-        <StakeBalances
-          title={t("stake.stakeBalanceTitle")}
-          dollarAmount={`${price}`}
-          osmoAmount={balance.toString()}
-        />
-        <StakeBalances
-          title={t("stake.rewardsTitle")}
-          dollarAmount="$26.89"
-          osmoAmount="20"
-        />
+    const summedStakeRewards = rewards?.reduce((acc, reward) => {
+      return reward.toDec().add(acc);
+    }, new Dec(0));
+
+    const coinPrettyStakeRewards = summedStakeRewards
+      ? new CoinPretty(osmo, summedStakeRewards)
+      : new CoinPretty(osmo, 0);
+
+    const dollarRewards =
+      priceStore.calculatePrice(coinPrettyStakeRewards) || "0";
+
+    const dollarBalance = balance ? priceStore.calculatePrice(balance) : 0;
+
+    const icon = (
+      <div className="flex items-center justify-center text-bullish-500">
+        <div className="mr-2 flex self-center">
+          <Icon id="open-book" height="14px" width="14px" />
+        </div>
+        <span className="caption text-sm">{t("stake.learn")}</span>
       </div>
-      <ValidatorSquadCard
-        setShowValidatorModal={setShowValidatorModal}
-        validators={validators}
-        usersValidatorsMap={usersValidatorsMap}
-      />
-      <div className="flex h-full w-full flex-grow flex-row space-x-2">
-        <RewardsCard
-          title={t("stake.collectRewards")}
-          titleIconUrl="www.google.com"
-        />
-        <RewardsCard
-          title={t("stake.investRewards")}
-          titleIconUrl="www.google.com"
+    );
+
+    return (
+      <GenericMainCard title={t("stake.dashboard")} titleIcon={icon} width="45">
+        <div className="flex w-full flex-row justify-between py-10">
+          <StakeBalances
+            title={t("stake.stakeBalanceTitle")}
+            dollarAmount={`${dollarBalance}`}
```

Comment:
I'm just using `priceStore.calculatePrice` here. I believe you can pass in a diff currency if desired, but the default is always USD. I'm not sure if anywhere in the app we are converting to anything other than USD or if we have plans to make things more dynamic in the future? My guess would be that if we can do account abstraction, we would allow people to change their default currency, in which case this call would stay the same but we would just add an extra parameter to the call. 

Perhaps the name is misleading, since though it is the `dollarBalance` here, a better name would be something like `fiatBalance`. The `$` here isn't 'dollar' but just the JS interpolation.

Reply:
I'm in this file anyway on my next PR, so I'll change the variable name there